hydra-main/src/main/java/com/addthis/hydra/job/web/JobRequestHandlerImpl.java

```diff
@@ -78,7 +78,7 @@ public Job createOrUpdateJob(KVPairs kv, String username, String token, String s
                     Splitter.on(',').omitEmptyStrings().trimResults().splitToList(kv.getValue("hosts", "")),
                     kv.getValue("minionType", DEFAULT_MINION_TYPE),
                     command, defaults);
-            updateOwnership(job, user);
+            updateOwnership(kv, job, user);
```

Comment:
Actually I think the best place to do this is before line 75 `job = spawn.createJob(` by setting the "creator" and "owner", and "group" entries in the `kv` if they are null or empty. Then you won't need the `updateOwnership` method anymore.